src/Generator/ParameterGenerator.php

```diff
@@ -163,14 +163,20 @@ public function __construct(
     }
 
     /**
-     * @param  string $type
+     * @param  mixed $type
```

Comment:
Seconding @Ocramius ; looking through the method body, this would be far simpler if it accepts a type instance.